### PR TITLE
Laboratory work #1, Aladyshkin Vasiliy - 20FPL3

### PR DESCRIPTION
**Pull request by froschalad**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
It's time to push some changes. October 1st is coming

**Comment by WhiteJaeger** [comment]
It's time to push some changes. October 1st is coming.

File: lab_1/main.py [hunk 1]
```diff
@@ -11,7 +11,16 @@ def tokenize(text: str) -> list or None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
-    pass
+    symbols = ['.', ',', '!', '*', '?', '-', '\\', '|', '/', ':', ';', '<', '>', '^', '@', '#', '$', '%', '&', '(',
+                   ')', '[', ']', '{', '}', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    if type(text) == str:
```
Comment (WhiteJaeger) [inline lab_1/main.py:16]: There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.

File: lab_1/main.py [hunk 2]
```diff
@@ -21,7 +30,11 @@ def remove_stop_words(tokens: list, stop_words: list) -> list or None:
     :param stop_words: a list of stop words
     :return: a list of tokens without stop words
     """
-    pass
+    if type(tokens) == list and type(stop_words) == list:
+        tokens = [token for token in tokens if token not in stop_words]
```
Comment (WhiteJaeger) [inline lab_1/main.py:34]: Good.

File: lab_1/main.py [hunk 3]
```diff
@@ -30,7 +43,17 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    freq_dict = {}
+    if type(tokens) == list:
```
Comment (WhiteJaeger) [inline lab_1/main.py:47]: There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.

File: lab_1/main.py [hunk 4]
```diff
@@ -40,7 +63,14 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
+    top_n_words = []
+    if type(freq_dict) == dict:
```
Comment (WhiteJaeger) [inline lab_1/main.py:67]: There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.

File: lab_1/main.py [hunk 5]
```diff
@@ -51,7 +81,12 @@ def create_language_profile(language: str, text: str, stop_words: list) -> dict
     :param stop_words: a list of stop words
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+    freq_dict = calculate_frequencies(remove_stop_words(tokenize(text), stop_words))
+    if type(language) == str and type(text) == str and type(stop_words) == list:
```
Comment (WhiteJaeger) [inline lab_1/main.py:85]: There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.

File: lab_1/main.py [hunk 6]
```diff
@@ -62,7 +97,17 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
     :param top_n: a number of the most common words
     :return: the distance
     """
-    pass
+    if type(unknown_profile) == dict and type(profile_to_compare) == dict and type(top_n) == int:
```
Comment (WhiteJaeger) [inline lab_1/main.py:100]: There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.

File: lab_1/main.py [hunk 7]
```diff
@@ -62,7 +97,17 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
...
+        top_of_unknown_profile = get_top_n_words(unknown_profile['frequency'], top_n)
+        top_of_profile_to_compare = get_top_n_words(profile_to_compare['frequency'], top_n)
+        shared_words = []
+        for word in top_of_unknown_profile:
+            if word in top_of_profile_to_compare:
+                shared_words.append(word)
+                distance = len(shared_words) / len(top_of_unknown_profile)
```
Comment (WhiteJaeger) [inline lab_1/main.py:107]: Are you sure about calculating the distance on each iteration of the `for word in top_of_unknown_profile:` loop?

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

**Comment by froschalad** [comment]
what is wrong with "unittests"?

File: lab_1/main.py [hunk 8]
```diff
@@ -51,7 +76,11 @@ def create_language_profile(language: str, text: str, stop_words: list) -> dict
     :param stop_words: a list of stop words
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+    if not isinstance((language, text), str) or not isinstance(stop_words, list):
```
Comment (WhiteJaeger) [inline lab_1/main.py:79]: ` isinstance((language, text), str)` does not work like that. `isinstance` supports only one object at a time. Split into several checks.

File: lab_1/main.py [hunk 9]
```diff
@@ -62,7 +91,17 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
     :param top_n: a number of the most common words
     :return: the distance
     """
-    pass
+    if not isinstance((unknown_profile, profile_to_compare), dict) or not isinstance(top_n, int):
```
Comment (WhiteJaeger) [inline lab_1/main.py:94]: `isinstance` supports only one object at a time. Split into several checks.

File: lab_1/main.py [hunk 10]
```diff
@@ -74,46 +113,12 @@ def detect_language(unknown_profile: dict, profile_1: dict, profile_2: dict, top
...
-
-
-def save_profile(profile: dict) -> int:
-    """
-    Saves a language profile
-    :param profile: a dictionary
-    :return: 0 if everything is ok, 1 if not
-    """
-    pass
+    if not isinstance((unknown_profile, profile_1, profile_2), dict) or not isinstance(top_n, int):
```
Comment (WhiteJaeger) [inline lab_1/main.py:116]: `isinstance` supports only one object at a time. Split into several checks.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

**Comment by WhiteJaeger** [comment]
I do not see any changes in `start.py`.

**Pull request closed without merging**